src/screens/Tasks/index.js

```diff
@@ -43,17 +46,32 @@ export default class Tasks extends Component {
                         ? {...task, completed: !task.completed }
                         : {...task}
         });
-        this.setState({tasks: tasksList});
+        this.state = {
+            task:[],
+        };
     }
 
+    componentDidMount(){
+        const self = this;
+        const {category} = this.state; 
+        const path = `${API_BASE_URL}/tasks/`;
+        fetch(path + category)
+            .then( data => data.json())
+            .then( data => {
+            const {tasks=[]} = data;
+            self.setState({tasks});
+        })
+    }
+ 
     render(){
         const { tasks = [], showAddTaskModal=false } = this.state;
+        const {category} = this.state;
```

Comment:
Ya habíamos hablado de esto, no necesitas el category en el state, es mejor que lo definas aquí:
`const category = this.props.navigation.getParam(category,'school');`